### PR TITLE
doc: add distribution page

### PR DESCRIPTION
**Pull request by rjbou**
As wiki is disabled, port distribution as part of opam.ocaml.org.
I updated versions information & some urls.
If maintainers (@mehdid @nbraud @avsm @pmetzger @xyproto @david-geiger @roptat @hannesm @TC01 @dra27) can take a look to check them, it'll be perfect. Like that, you'll also know about this page and be able to update it on further releases.

Thanks for all the porting work!

### CODE REVIEW

**Comment by hannesm** [comment]
I'm not entirely sure about that table in some aspects: "latest opam version" quickly gets out of date (and you already added hyperlinks to the authoritative version information sources); and the ordering - I'd appreciate a lexicographic order for quick browsing; I don't really understand "Official distribution packages" vs "Others" (where the first is "opam packaged in operating systems" -- `distribution` is a term only used in Linux afaik)

The maintainer information also gets stale, and is usually part of the package in the operating system package system, and imho could be dropped.

**Comment by roptat** [comment]
I'm good with the content for Guix.

I think official distribution packages are packages provided as part of the official distribution list of packages, whereas others are packaged as part of a third-party repository only (like a ppa, the aur, copper, ...). I agree about the lexicographic ordering. I'm more reserved with version and maintainer information, because they are still good to have, not only for users, but also for opam developpers if they need to contact someone from a distro.

I'm also wondering if you should rather merge this information with https://opam.ocaml.org/doc/Install.html?

File: doc/pages/Distribution.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+| Ubuntu Linux (19.04,cosmic)   | 1.2.2       | -                                                                                                                 | [Package search](http://packages.ubuntu.com/search?keywords=opam&searchon=names&suite=all&section=all)
+| Ubuntu Linux (19.04,disco)    | 2.0.3       | -                                                                                                                 | [Package search](http://packages.ubuntu.com/search?keywords=opam&searchon=names&suite=all&section=all)
+| Ubuntu Linux PPA              | 2.0.4       | Anil Madhavapeddy [@avsm](https://www.github.com/avsm)                                                            | [Anil's official OPAM PPA](https://launchpad.net/~avsm) and [ppa post](https://discuss.ocaml.org/t/opam-2-0-experimental-ppas/2446)
+| Homebrew (MacOS X)            | 2.0.0       | Anil Madhavapeddy [@avsm](https://www.github.com/avsm)                                                            |
```
Comment (avsm) [inline doc/pages/Distribution.md]:
```suggestion
| Homebrew (MacOS X)            | 2.0.4       |                                                            |
```

updated version, and homebrew has no official maintainer system I think, so removing my name

File: doc/pages/Distribution.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+| NixOS                         | 1.2.2/2.0.0 | Henry Till                                                                                                        | [Package definitions](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/tools/ocaml/opam)
+| Gnu Guix                      | 2.0.4       | Julien Lepiller [@roptat](https://github.com/roptat)                                                              | [Package definition](https://git.savannah.gnu.org/cgit/guix.git/tree/gnu/packages/ocaml.scm#n428)
+| FreeBSD                       | 2.0.4       | Hannes Mehnert [@hannesm](https://www.github.com/hannesm)                                                         | [Package search](http://www.freebsd.org/cgi/ports.cgi?query=opam&stype=all)
+| OpenBSD                       | 1.2.2       | Anil Madhavapeddy [@avsm](https://www.github.com/avsm)                                                            | [Package page](http://ports.su/sysutils/opam,-main)
```
Comment (avsm) [inline doc/pages/Distribution.md]:
```suggestion
| OpenBSD                       | 2.0.4       | Christopher Zimmerman (@madroach)                                                            | [Package page](http://ports.su/sysutils/opam,-main)
```

**Comment by xyproto** [comment]
The current link to the Arch Linux opam package will also show other packages with "opam" in the name, if new packages are added in the future.

Here is an alternative URL that links directly to the [Arch Linux opam package](https://www.archlinux.org/packages/community/x86_64/opam/). (Arch Linux currently only supports 64-bit x86, distros like Arch Linux ARM are separate distros).

Oh, and I'm in favor of sorting the entries in alphabetical order...

Other than that, it looks good to me. :+1:

**Comment by rjbou** [comment]
Lexicographic order updated.

@hannesm Originally, tables were split to differentiate official & third party packages, but I won't insist to keep it that way. If it's more convenient to have an unique table, it's ok for me.

@roptat This page will be available directly via the opam.ocaml.org/doc menu, the item is added under `Packaging`.

@xyproto url updated!

**Comment by rjbou** [comment]
@hannesm I forgot one thing, even if information is reachable by following links, having them aggregated in one place is better for an overview, of opam availability accross platform & also current maintainers to keep a link with them (repology gives the first part of the information).

**Comment by pmetzger** [comment]
If this is what the final page looks like, it looks pretty bad; maybe there's some formatting missing?

https://opam.ocaml.org/doc/Distribution.html

**Comment by rjbou** [comment]
@pmetzger  yes, it's because omd doesn't handle tables, which are in extended markdown.

**Comment by avsm** [comment]
@rjbou if you create an omd issue, @clecat could take a look at that as he's hacking on a rewrite in the master branch with @nojb atm

**Comment by rjbou** [comment]
related issue ocaml/omd#188

**Comment by pmetzger** [comment]
Even if it doesn't handle tables, maybe some other way of formatting the page without tables could be used temporarily?

**Comment by rjbou** [comment]
Yes, it is in the referenced PR :)

**Pull request merged**